resources/lib/comm.py

```diff
@@ -33,31 +33,30 @@
 
 def fetch_url(url, token=None):
     """
-        Simple function that fetches a URL using urllib2.
+        Simple function that fetches a URL using requests.
         An exception is raised if an error (e.g. 404) occurs.
     """
     utils.log("Fetching URL: %s" % url)
-
-    # Token headers
-    headers = {}
-    if token:
-        headers = {'x-media-mis-token': token}
-
-    request = urllib2.Request(url, headers=headers)
-    return urllib2.urlopen(request).read()
-
-
-def fetch_token():
+    with requests.Session() as session:
+        # Token headers
+        headers = {}
+        if token:
+            update_token(session)
+    
+        request = session.get(url, headers=headers)
+        data = request.text
+    return data
+
+
+def update_token(session):
     """
         This functions performs a HTTP POST to the token URL
-        and it will return a token required for API calls
+        and it will update the requests session with a token 
+        required for API calls
     """
-    req = urllib2.Request(config.TOKEN_URL, '')
-    res = urllib2.urlopen(req)
-    json_result = json.loads(res.read())
-    res.close()
-
-    return json_result['token']
+    res = requests.post(config.TOKEN_URL)
+    json_result = json.loads(res.text)
```

Comment:
Would be nice to be a little defensive here and wrap in a try except with a nicer message than failed to decode json?